### ISSUE TITLE
[dev-go/go-bindata] Repository switch

### ISSUE THREAD

**Issue by Dr-Terrible** [Dr-Terrible/go-overlay#76]
Apparently, the original maintainer of [go-bindata](https://github.com/jteeuwen/go-bindata) is no more active¹ on GitHub since 2015. As a result, `go-bindata` has been forked by [several people](https://github.com/jteeuwen/go-bindata/issues/165#issuecomment-347415857) in attempt to fix the overwhelming number of pending issues and bugs.

Unfortunately, none of those forks are API compatible with the original package, or with each others, leaving behind a hell to deal with for distro's packagers.

Unless this situation improve drastically in the next days, I'm going to remove `dev-go/go-bindata` from this overlay and rewrite the few ebuilds depending on it to rely on a bundled copy. People wishing to use the tool can still install it from portage.

¹- he is active on other web sites, so he is well and sound and I'm glade of that.

**Comment by Dr-Terrible** [comment]
I settled with Kevin Burke's [fork](https://github.com/kevinburke/go-bindata).

That fork is API compatible with the original package and the author doesn't wish to diverge from it, or add new incompatible features; as a bonus, even [rsc](https://github.com/rsc) is contributing fixes. So far so good, it looks promising.

I'll keep this report open for a while, and for further feedback from other collaboratos.

**Comment by Dr-Terrible** [comment]
The drama surrounding `go-bindata` keeps [raging on](https://donatstudios.com/GithubsTotalSecurityFacepalm). Jim Teeuwen removed himself from GitHub, together with the original `go-bindata` repository; [someone recreated](https://github.com/jteeuwen/go-bindata/issues/5) both the GitHub account and the repository based on a fork of the original project. Meh, what a mess!

My choice to stick with Kevin Burke's fork turned out to be wise: only one package is affected `www-apps/beehive`.

**Comment by ilovezfs** [comment]
@Dr-Terrible any opinion on
https://github.com/shuLhan/go-bindata vs
https://github.com/kevinburke/go-bindata

?

I just created a Homebrew formula for go-bindata using https://github.com/kevinburke/go-bindata based on the success you described above. The conversation here https://github.com/containous/traefik/issues/2965 also brought up https://github.com/shuLhan/go-bindata.

**Comment by Dr-Terrible** [comment]
@ilovezfs Hi, originally I used `shuLhan/go-bindata` as an alternative to `jteeuwen/go-bindata`. As a result I got bit by a plethora of _"fatal error: runtime: out of memory"_ from pretty much every packages that internally relied on some old snapshot taken from `jteeuwen/go-bindata`; as an example see this report of mine: muesli/beehive#175 (which I'm going to properly fix it soon).

shuLhan's fork may be OK-_ish_ for people writing new packages in Go, but when packaging older ones there is a high chance that the compilation may fails when coupled with recent versions of the Go's toolchain — most notably with Go v1.9+. So far, kevinburke's fork was the only one that allowed me to avoid the latter without forcing me to fix upstream bugs; so it was a win-win situation for me.

To be honest, quite some time has passed since I tested shuLhan's fork ([v3.2.0](https://github.com/Dr-Terrible/go-overlay/commits/master/dev-go/go-bindata)), so I'm going to try out v3.3.0, which was released a few days ago, and see if that snapshot fixes all the _runtime failures_ or if it make things even worse. Then I'll report here my feedback. It's not a big deal :+1:

**Comment by ilovezfs** [comment]
@Dr-Terrible cool, thank you! It would be great if those forks could merge somehow, I think.

**Comment by Dr-Terrible** [comment]
> @Dr-Terrible cool, thank you! It would be great if those forks could merge somehow, I think.

Indeed, a merge from both shuLhan's and kevinburke's forks would be great. I honestly don't understand why there is such a strong territoriality around `go-bindata` (here I'm talking about the people who recreated jteeuwen's repo instead to ask help from the Go community for a proper fix). It's just silly and childish :D In the end everyone was hurt by it. But I'm digressing.

**Comment by Dr-Terrible** [comment]
Well, shuLhan/go-bindata v3.3.0 has made a lot of progress compared to the previous release (v3.2.0) as recap in the table below (I'm using Go v1.10 for the tests):

```
  Legend:
  YES → package compiles AND works as expected
  NO  → package doesn't compile OR doesn't work as expected
```

package¹ | shuLhan/go-bindata (v3.2.0)  | shuLhan/go-bindata (v3.3.0) | kevinburke/go-bindata (v3.7.0)
------------ | ------------- | ------------- | ---------
[pgweb](http://sosedoff.github.io/pgweb) | YES | YES | YES
[gcli](https://github.com/tcnksm/gcli) | YES | YES | YES
[go-fuzz](https://github.com/dvyukov/go-fuzz) | NO | YES | YES
[gotests](https://github.com/cweill/gotests) | NO | YES | YES
[gogs](https://gogs.io) | NO | YES **²** | YES
[go-swagger](https://goswagger.io) | NO | YES | YES
[dex](https://github.com/coreos/dex) | NO | YES | YES
[beehive](https://github.com/muesli/beehive) | NO | **YES** | YES
[go-bindata-assetfs](https://github.com/elazarl/go-bindata-assetfs) | NO | NO | YES
[xo](https://github.com/knq/xo) | NO | NO | YES

¹- these are all packages (from this repository) that, one way or another, explicitly require an external invocation of go-bindata executable;
²- package compiles but shows random run-time failures when accessing binary data;

The biggest offender (beehive's memory failures) has been fixed, kudos to shuLhan for the effort. Still, kevinburke's fork has a slightly higher successful rate. Obviously, my sample size is too little to draw any conclusive comparison and I would not be surprised if, in the future, shuLhan/go-bindata v3.4.0 will show comparable results to kevinburke's fork.

So far, shuLhan/go-bindata v3.3.0 has some issues handling compilation with CGO enabled.
